src/overdub.jl

```diff
@@ -491,17 +495,52 @@ const OVERDUB_FALLBACK = begin
     code_info
 end
 
+"""
+    ReflectOn{Tuple{F, ArgTypes...})
+
+When used in place of `f` in `overdub(ctx, f, g, args...)`, causes the method
+of the function of type `F` with method signature `ArgTypes` to be overdubbed
+and called with `args`. `g` is used as `#self#`, the function itself.
+
+It is assumed that the method body will work with `args` even though they may
+not be the same type prescribed by the original method signature. Useful when
+writing passes which you to extract the code for a base type and rewrite it to
+work on a custom type.
+
+```julia
+julia> Cassette.@context Foo
+
+julia> foo(x::Float64) = "float"
+
+julia> foo(x::Int) = "int"
+
+julia> Cassette.overdub(Foo(), Cassette.ReflectOn{Tuple{typeof(foo), Int64}}(), foo, 1.0)
+"int"
+
+julia> Cassette.overdub(Foo(), Cassette.ReflectOn{Tuple{typeof(foo), Float64}}(), foo, 1)
+"float"
+```
+"""
+struct ReflectOn{T<:Tuple}
```

Comment:
I think for API consistency, it should be `ReflectOn{f, Tuple{ArgTypes...}}` instead of `ReflectOn{Tuple{f, ArgTypes...}}`

Reply:
fair enough.